server/url_params.go

```diff
@@ -63,6 +67,77 @@ func ExtractParametersFromUrl(url *url.URL) (params URLParameters, err error) {
 		params.pref.Builders = targetBuildersQuery
 	}
 
+	refundAddressQuery, ok := url.Query()["refund"]
+	if ok {
+		if len(refundAddressQuery) == 0 {
+			return params, ErrIncorrectRefundQuery
```

Comment:
**nitpick (non-blocking):** It would be nice if we had more granular error message, e.g.

```suggestion
			return params, ErrEmptyRefundAddressQuery
```